app/src/common/shared/org/mozilla/servo/ServoDisplay.java

```diff
@@ -0,0 +1,36 @@
+package org.mozilla.servo;
+
+import android.view.Surface;
+
+import org.mozilla.geckoview.GeckoDisplay;
+import org.mozilla.geckoview.GeckoSession;
+
+public class ServoDisplay extends GeckoDisplay {
+
+    private ServoSession mServoSession;
+
+    public ServoDisplay(final GeckoSession session) {
+        super(session);
+        mServoSession = (ServoSession) session;
+    }
+
+    @Override
+    public void surfaceChanged(final Surface surface, final int width, final int height) {
+        mServoSession.onSurfaceReady(surface, width, height);
+    }
+
+    @Override
+    public void surfaceDestroyed() {
+        mServoSession = null;
```

Comment:
on GeckoView `surfaceDestroyed` pauses the compositor internally. We should so something similar in Servo. It's used to pause the compositor while in WebVR, or also when the activity goes to background.